source/Framework Basics.rst

```diff
@@ -0,0 +1,173 @@
+Framework Basics
+=================
+
+Interventions
+-------------
+Interventions can help you execute specific actions based on the current state of the simulation. The simulation engine of Bharat-Sim checks if an inactive intervention can be activated based on the given conditions during every tick. All the active interventions will be recorded in the context and the user can query the context using ``context.activeInterventionNames.contains(interventionName)``.
+
+There are 4 different kinds of Interventions available in Bharat-Sim:
+
+1.Intervention
+~~~~~~~~~~~~~~~~~~~~~
+This is a Generic intervention that can be invoked several times or once when the activation condition is met based on the parameters passed.
+This can be used to implement lockdows during an epidemic everytime the infected fraction of agent reaches a certain threshold.
+There are 5 parameters that can be passed ``Intervention`` object:
+
+    * ``interventionName``: This is the unique name of the intervention.
+    * ``shouldActivateFunc``: This function which tells whether this intervention should be activated or not. The function should return a Boolean value.
+    * ``shouldDeactivateFunc``: This function which tells whether this intervention should be deactivated. The function should return a Boolean value.
+    * ``firstTimeActionFunc``: This is an optional function which will be executed at the start of the intervention
+    * ``whenActiveActionFunc``: This is an optional function which will be executed per tick when intervention is active.
+
+Example using the ``Intervention`` object:
+````````````````````````````````````````````````````
+.. code-block:: scala
+
+    //This implements a lockdown when the nuber of agents in the infected compartment is greater than or equal to 2000
+    private def addLockdown(implicit context: Context): Unit = {
+
+    var interventionActivatedAt = 0
+    val interventionName = "lockdown"
+    val activationCondition = (context: Context) => getInfectedCount(context) >= 2000
```

Comment:
What is the `getInfectedCount` function? A beginner might not understand that this is not a framework defined function. Explain that this is user-defined, and you can link to the "Writing your first program" section and say that it will be explained in greater detail there.